packages/payment-detection/README.md

```diff
@@ -1,22 +1,34 @@
 # @requestnetwork/payment-detection
 
 `@requestnetwork/payment-detection` is a typescript library part of the [Request Network protocol](https://github.com/RequestNetwork/requestNetwork).
-It contains client-side payment detection for all supported payment networks.
+It contains the implementation of request-related events interpretations, typically onchain payments of requests.
 
-### Payment and Refund detections
+The interpretation of events is specified by the payment extension added to the request. [Cf. advanced-logic specifications](../advanced-logic/specs/).
+
+# Balance and events
 
 If a payment network has been given to the request, the payment detection can be done.
 
-From the information provided in payment network, the library will feed the property `balance` of the request with:
+Based on information found in the payment network state, and included manual payment declarations, the library will perform queries and feed the property `balance` of the request with:
+
+- `balance`: the detected amount paid on the request, in request currency
+- `events`: all the payments, payment declarations, refunds, and other balance-related events with the amount, timestamp etc...
+
+# Design: retrievers and detectors
+
+This library relies on two concepts:
+
+- **Retrievers** perform RPC calls or TheGraph and fetch relevant events. Balance-impacting events are fetched with `InfoRetrievers`, implementing the `getTransferEvents()` method (cf. [IPaymentRetriever](./src/types.ts))
+- **Payment detectors** call retrievers and interpret events according to the payment network (cf. [Abstract PaymentDetectorBase](./src/payment-detector-base.ts))
+
+## Subgraph-based payment retrievers
+
+For TheGraph-based information retrieval, we implement a generic method `getTheGraphClient()` in `./src/thegraph/index.ts`, based on the queries in `/src/thegraph/queries` and automated types.
```

Comment:
what do you mean by generic here? In TS generic usually refer to `function foo<T>(){...}`. 
I don't this this method is actually very "generic" anyway, as it's only exposing a little set of methods, generated based on the queries
```suggestion
For TheGraph-based information retrieval, a client can be retrieved using `getTheGraphClient()` in `./src/thegraph/index.ts`. It provides a strongly typed interface, generated based on the queries in `/src/thegraph/queries`.
```